packages/salesforcedx-sobjects-faux-generator/src/generator/fauxClassGenerator.ts

```diff
@@ -267,47 +268,38 @@ export class FauxClassGenerator {
     }
     this.cleanupSObjectFolders(sobjectsFolderPath, SObjectCategory.STANDARD);
 
-    const connection = await Connection.create({
-      authInfo: await AuthInfo.create({
-        username: await ConfigUtil.getUsername(projectPath)
-      })
-    });
-    const describe = new SObjectDescribe(connection);
-
     if (
       this.cancellationToken &&
       this.cancellationToken.isCancellationRequested
     ) {
       return this.cancelExit();
     }
 
-    let fetchedSObjects: SObject[] = [];
-    try {
-      fetchedSObjects = await describe.fetchObjects(startupMinSObjects);
-    } catch (errorMessage) {
-      return this.errorExit(
-        nls.localize('failure_in_sobject_describe_text', errorMessage)
-      );
-    }
-
-    const standardSObjects: SObject[] = [];
-    // tslint:disable-next-line:prefer-for-of
-    for (let i = 0; i < fetchedSObjects.length; i++) {
-      standardSObjects.push(fetchedSObjects[i]);
+    if (!this.createIfNeededOutputFolder(standardSObjectsFolderPath)) {
+      throw nls.localize('no_sobject_output_folder_text', standardSObjectsFolderPath);
     }
 
-    this.result.data.standardObjects = standardSObjects.length;
-    this.result.data.customObjects = 0;
-
-    this.logFetchedObjects(standardSObjects, []);
+    const sobjectDecl: SObjectDefinition[] = this.getSObjectSubsetDefinitions();
+    this.generateAndWriteFauxClasses(sobjectDecl, standardSObjectsFolderPath);
+    this.result.data.standardObjects = sobjectDecl.length;
+    this.logSObjects('Standard', sobjectDecl.length);
+    return this.successExit();
+  }
 
-    try {
-      this.generateFauxClasses(standardSObjects, standardSObjectsFolderPath);
-    } catch (errorMessage) {
-      return this.errorExit(errorMessage);
+  // VisibleForTesting
+  public generateAndWriteFauxClasses(sobjectDecl: SObjectDefinition[], standardSObjectsFolderPath: string) {
```

Comment:
What's the benefit of adding this new method instead of using the existing `generateFauxClasses` used in the normal generator flow ?

Reply:
The generateFauxClasses takes SObject array as input and to generate that we would need the earlier 2M definition file.  If we go one level more specific, as what generateAndWriteFauxClasses here requires, we should be fine with the simpler declarations we have in the json now.

Reply:
Got it, can you add a comment about this at the top of the method ?